### DISCUSSION TITLE
Show the nested document structure in the url eg /doc/clients/acme/systems-OUKKGX9GAK

### DISCUSSION THREAD

**Discussion by brendanheywood** [outline/outline#6965] [Ideas]
Lets say we have a collection for clients:

/collection/clients-71Fv7gknlR

and under that a client for 'Acme'

/doc/acme-MwaVdkpPZe

and then lets say we have a page under that:

/doc/something-UyhOnTdgDA

Each url doesn't show the structure of where we are. My understanding of the url routing is that most of the url is fairly irrelevant and its only the bit at the end which identifies the pages eg 'MwaVdkpPZe' so we could easily include lots of contextual information into the url:

Either something very minimal like this:
```
/clients-71Fv7gknlR
/clients/acme-MwaVdkpPZe
/clients/acme/contracts-UyhOnTdgDA
/clients/acme/systems-OUKKGX9GAK
```

Or if we really need prefixes to simplify the url routing under the hood then:

```
/collection/clients-71Fv7gknlR
/doc/clients/acme-MwaVdkpPZe
/doc/clients/acme/contracts-UyhOnTdgDA
/doc/clients/acme/systems-OUKKGX9GAK
```

We are about to embark on a large wiki migration and the loss of this visual structure is a big downside. We have a large number of similar pages for different clients or systems and they will have similar naming conventions so we need to tell them apart easily.

If there was general agreement on the concept we could contribute this.

**Comment by tommoor** [comment]
Hi Brendan, it's an interesting idea – but I'm not sure the upsides outweigh the downsides with this one. I believe we want to lean more into documents being flexible in their location, potential existing in multiple places rather than being fixed in the structure.

I'm interested in what scenario you're relying on paths? You mentioned similar pages for different clients – one thing we could definitely do is introduce the collection name / breadcrumb in the tab title.

**Comment by brendanheywood** [comment]
> I'm interested in what scenario you're relying on paths?

This would be purely to improve the UX of handling the urls without being able to preview them or visit them. Like embedding the url in another system, document, email, chat etc where a preview isn't loadable. So adding it to the html title doesn't help as you can't see the document yet.

I'm not sure I see any downsides personally, as its all bonus info which has no routing meaning so is just ignored. Lots of routing frameworks work like this, eg stack overflow:

```
https://stackoverflow.com/questions/78555926/round-half-down-in-informix
https://stackoverflow.com/questions/78555926/some-random-stuff-I-made-up
```

Some quick testing show that outline already works this way, eg if I have a page like this:

`doc/some-test-page-v7Fwt48lz9`

and if I hack the url then `doc/some-stuff-i-made-up-v7Fwt48lz9` also works (but redirects).

but I note that this does not work `/doc/grandparent/parent/some-test-page-v7Fwt48lz9`

right now this does work `/doc/grandparent-parent-some-test-page-v7Fwt48lz9` except for the redirect which is 90% close

So it seems to me that changing the routing so that the canonical url includes the extra context is a pretty small change.

It doesn't even need to be `/` as the delimiter. The point of using / in most routing frameworks is so you can remove a path and navigate to the parent page, but this won't work anyway because it is missing the v7Fwt48lz9 id for the parent and grandparent page. It could be any other character eg 

`/doc/grandparent>parent>some-test-page-v7Fwt48lz9`


If a document exists in multiple locations that can also work fine, you effectively end up with multiple urls which serve almost identical content. When you actually view the document you are going to show the crumb trail in that page, so if it has multiple locations then would you simultaneously show two crumb trails? Or would you effectively have two pages with different contexts and each has a single crumb trail (this is what I would expect).

Taking an example from flickr you could have a bunch of different urls all with different embedded context. They all serve the same content but with different navigation context:

```
https://www.flickr.com/photos/brendan_heywood/53588150802
https://www.flickr.com/photos/brendan_heywood/53588150802/in/photostream/
https://www.flickr.com/photos/brendan_heywood/53588150802/in/album-72177720312761351/
```

> You mentioned similar pages for different clients – one thing we could definitely do is introduce the collection name / breadcrumb in the tab title.

As this is about the UX of handling the urls themselves, we haven't yet viewed the document or gotten a preview so the tab title isn't known yet. In our case I'm not sure the collection name would help as I suspect we'll have one collection for Clients (or whatever) and then a fairly deeply nested set of pages under that. I don't think it would be useful to have the complete crumb trail in the html title as it would clutter things up. Adding the Collection I think does have merit for other reasons outside of this issue, and if it was added I'd make it a suffix so that truncated tab names show the most important part.
